spec/requests/profile_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+describe "Profile", type: :request do
+  describe "#index" do
+    context "when user is not logged in" do
+      before { get profile_index_path, params: {uid: "user-id"} }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when user is logged in" do
+      let(:user1) { create(:user1) }
+      before { sign_in user1 }
+      it "renders index template" do
+        get profile_index_path, params: {uid: user1.uid}
+        expect(response).to render_template(:index)
+      end
+
+      context "when profile belongs to logged in user" do
+        it "assigns profile instance to be that of logged in user" do
+          get profile_index_path, params: {uid: user1.uid}
+          expect(assigns(:profile)).to eq(user1)
+        end
+
+        context "when user has no moments and strategies" do
+          it "assigns stories instance with an empty array" do
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array([])
+          end
+        end
+
+        context "when user has moments or strategies" do
+          it "assigns stories instance with user moments or strategies" do
+            stories = [build_stubbed(:moment), build_stubbed(:strategy)]
+            allow_any_instance_of(ProfileController)
+              .to receive(:get_stories).and_return(stories)
+            get profile_index_path, params: {uid: user1.uid}
+            expect(assigns(:stories)).to match_array(stories)
+          end
+        end
+      end
+
+      context "when profile does not belong to user" do
+        let(:profile_owner) { build_stubbed(:user2) }
+
+        before do
+          allow(User).to receive(:find_by).and_return(profile_owner)
+          get profile_index_path, params: {uid: profile_owner.uid}
+        end
+
+        it "does not assign stories" do
+          expect(assigns(:stories)).to be_nil
+        end
+
+        it "assigns profile instance to profile owner" do
+          expect(assigns(:profile)).to eq(profile_owner)
+        end
+      end
+    end
+  end
+
+  describe "#data" do
+    let!(:user) { create(:user) }
```

Comment:
```suggestion
    let(:user) { create(:user) }
```

Is the `!` needed here? I'm assuming we have `!` on moment because a moment is a story, right? We would need that to be available before we go on the page itself to make sure that the story exist or something?

On the other hand, I only see it used during 

```ruby
it "returns a response with the correct path" do
```

We can probably move it in an Arrange step setup for that test.